src/Provider/Nominatim/Model/NominatimAddress.php

```diff
@@ -54,6 +54,16 @@ final class NominatimAddress extends Address
      */
     private $type;
 
+    /**
+     * @var array|null
+     */
+    private $addressData;
```

Comment:
I would rename `$this->addressData` (and related functions) to `$this->details`.